report/3.Theory/SemanticsDescription.tex

```diff
@@ -451,27 +505,29 @@ \subsection{Big-Step Semantic for Expressions}
 
 \subsection{Big-Step Semantic for Statements}
 Statements $Stm$ can change the store, because they can change the value of an existing variable.
-The transition-system for $stm$ is $(Stm\ \Cross\ Sto\ \cup Sto, \rightarrow_{stm}, Sto)$.
-Given the variable bindings $env_V$ and the procedure bindings $env_P$, the execution of the statement $S$ will 
+Because of the fact that the category have rewriting rules for declarations, this means that statements can also both change the variable environment $Env_V$ and the function environment $Env_F$.
+The transition-system for $stm$ is:
+\begin{center}
+	$((Stm\ \Cross\ Env_V\ \Cross\ Env_F\ \Cross Sto)\ \cup Env_V\ \Cross\ Env_F\ \Cross\ Sto, \rightarrow_{stm}, Env_V\ \Cross\ Env_F\ \Cross\ Sto)$
+\end{center}
+Given the variable bindings $env_V$ and the function bindings $env_F$, the execution of the statement $S$ will 
 result in $sto$ changing to $sto'$.
-the formal parameters for the procedure calls in the transitions below, are denoted with $x$.
+the formal parameters for the function calls in the transitions below, are denoted with $x$.
 
 This means that the transitions must be on the form.
 
 \begin{center}
 	\begin{math}
-		{env_V, env_P \vdash \langle S, sto \rangle \rightarrow_{stm} sto'}
+		{\langle S, env_V, env_F, sto \rangle \rightarrow_{stm} (sto', env_V', env_F')}
```

Comment:
Jeg er enig med denne regel, men du følger den ikke i alle de andre regler.

For eksempel, i Assignment reglen går transitionen til sto[l -> v] i stedet for til (sto[l -> v], env_V, env_F)

Selvom environments ikke bliver ændret i alle regler, skal de stadig optræde (uændret) på højre side af transitionspilen.

Reply:
Ja true. glemte dem lige. Fixed